lib/rex/post/sql/ui/console/command_dispatcher/client.rb

```diff
@@ -137,11 +137,31 @@ def cmd_query(*args)
                 end
 
                 result = run_query(args.join(' '))
+                case result[:status]
+                when :success
+                  # When changing a database in MySQL, we get a nil result back.
+                  # When changing a database in MSSQL, we get a result, but it doesn't contain colnames or rows.
+                  normalised_result = normalise_sql_result(result[:result])
+
+                  # MSSQL returns :success, even if the query failed due to wrong syntax.
+                  if normalised_result[:errors].any?
+                    print_error "Query has failed. Reasons: #{normalised_result[:errors].join(', ')}"
+                    return
+                  end
 
-                normalised_result = result[:status] == :success ? normalise_sql_result(result[:result]) : result[:result]
-                formatted_result = format_result(normalised_result)
+                  if normalised_result[:rows].nil? || normalised_result[:columns].nil? # Should this be AND?
```

Comment:
Did you happen to figure out if this should have been an AND?

Reply:
Calling it out as I seen the comment 😄

Reply:
Will keep this as an OR in an upcoming commit. Rex Table throws an exception if nil values are provided, so we need an OR here 👍